src/Microsoft.DotNet.CodeFormatter.Analyzers/OptimizeNamespaceImportsOptions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+
+using Microsoft.CodeAnalysis.Options;
+ 
+namespace Microsoft.DotNet.CodeFormatter.Analyzers
+{
+    [Export(typeof(IOptionsProvider)), Shared]
+    internal class OptimizeNamespaceImportsOptions : IOptionsProvider
+    {        
+        public IEnumerable<IOption> GetOptions()
+        {
+            return new List<IOption>
+            {
+                Enabled,
+                RemoveUnnecessaryImports,
```

Comment:
We don't support all these options, obviously. I've filled out a prospective set that we might support in the analyzer, to give a better sense of what this pattern looks like in practice.